guides/advanced_user_guide.md

```diff
@@ -70,7 +71,11 @@ Here, the `apiVersion`, `kind`, and `name` are mandatory fields for all custom t
       - **--apiVersion**: Kubernetes API Version for your custom task CRD.
       - **--kind**: Kubernetes Kind for your custom task CRD.
       - **--name**: Kubernetes Resource reference name for your custom task CRD.
-      - **--taskSpec** (optional): Kubernetes Resource Spec for your custom task CRD. The value needs to define in Python Dictionary.
+      - **--taskSpec** (optional): Kubernetes Resource Spec for your custom task CRD. One of `--taskSpec` or `--taskRef` can be specified at a time.
+        The value should be a Python Dictionary.
+      - **--taskRef** (optional): Kubernetes Resource Spec for your custom task CRD. This gets inlined in the pipeline.  One of `--taskSpec` or `--taskRef` can be specified at a time.
```

Comment:
`taskSpec` should be the one that inlined in the pipeline. `taskRef` just generate the spec in a separate CRD.

Reply:
Ok, I was thinking, if we swap the taskSpec meaning, the custom controller who do not yet support embedded spec, will stop working.

Reply:
Updated.